spec/custom/index.html

```diff
@@ -584,167 +459,229 @@
+        </ol>
+
+        <p class="note">This occurs when author script constructs the custom element directly, e.g. via <code>new MyCustomElement()</code>.</p>
+    </li>
+
+    <li>Let <var>instance</var> be the last entry in <var>definition</var>'s <a href="#dfn-element-definition-construction-stack">construction stack</a>.</li>
+
+    <li>
+        <p>If <var>instance</var> is an <a><i>already constructed</i></a> marker, <a href="https://heycam.github.io/webidl/#dfn-throw">throw</a> an <code>InvalidStateError</code> and abort these steps.</p>
+
+        <p class="note">This can occur when the author code inside the <a>custom element constructor</a> invokes <code>super()</code> multiple times.</p>
+    </li>
+
+    <li>Replace the last entry in <var>definition</var>'s <a href="#dfn-element-definition-construction-stack">construction stack</a>a with an <a><i>already constructed</i></a> marker.</li>
+
+    <li>Return <var>instance</var>.</li>
 </ol>
-</div>
+
+<p class="note">For now, the <a href="https://html.spec.whatwg.org/multipage/dom.html#htmlelement"><code>HTMLElement</code></a> constructor cannot be invoked directly. It only works when used via a <code>super()</code> call inside a <a>custom element constructor</a>.</p>
+
+</section>
+
 <section id="extensions-to-document-interface-to-instantiate">
-<h3>Extensions to <a href="https://dom.spec.whatwg.org/#document"><code>Document</code></a> Interface</h3>
+<h3>Changes to <a href="https://dom.spec.whatwg.org/#document"><code>Document</code></a> Methods</h3>
 
-<p>To allow creating both <a>custom tag</a> and <a>type extension</a>-style <a data-lt="custom element">custom elements</a>, the <a href="https://dom.spec.whatwg.org/#dom-document-createelement"><code>createElement</code></a> or <a href="https://dom.spec.whatwg.org/#dom-document-createelementns"><codE>createElementNS</code></a> methods have <a href="https://heycam.github.io/webidl/#idl-overloading">overloads</a> with a <code>typeExtension</code> argument:</p>
+<p>To allow creating both <a>custom tag</a> and <a>type extension</a>-style <a data-lt="custom element">custom elements</a>, the <a href="https://dom.spec.whatwg.org/#dom-document-createelement"><code>createElement</code></a> or <a href="https://dom.spec.whatwg.org/#dom-document-createelementns"><codE>createElementNS</code></a> methods gain optional <var>typeExtension</var> arguments. Their new definitions are:</p>
 
 <pre class='idl'>
 partial interface Document {
-    Element createElement(DOMString localName, DOMString typeExtension);
-    Element createElementNS(DOMString? namespace, DOMString qualifiedName, DOMString typeExtension);
+    Element createElement(DOMString localName, optional DOMString typeExtension);
+    Element createElementNS(DOMString? namespace, DOMString qualifiedName, optional DOMString typeExtension);
```

Comment:
Should we go for a dictionary as final argument to allow future extensions without getting awkward four-argument methods? I think I'd prefer that.